srml/contract/src/lib.rs

```diff
@@ -148,8 +151,42 @@ decl_module! {
 	/// Contracts module.
 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
 		fn deposit_event<T>() = default;
-		// TODO: Change AccountId to staking::Address
-		/// Make a call to a specified account, optionally transferring some balance.
+
+		/// Updates the schedule for metering contracts.
+		///
+		/// The schedule should have a greater version than the stored schedule.
+		fn update_schedule(schedule: Schedule<T::Gas>) -> Result {
+			if <Module<T>>::current_schedule().version >= schedule.version {
+				return Err("Schedule should have a greater version");
```

Comment:
```suggestion
				return Err("Schedule version must be equal or greater");
```

Reply:
hmm, I think the suggestion is not correct: the incoming schedule's version should be strictly greater than the currently saved one.

Reply:
You are right. Would still rephrase.

```suggestion
				return Err("new schedule must have a greater version than current");
```

Reply:
Then you need to change `<Module<T>>::current_schedule().version >= schedule.version` to `<Module<T>>::current_schedule().version > schedule.version`

Reply:
Hm it seems like this is not correct... Can you check my logic?

- in order to run contract code we need to instrument it with gas metering. To do that we need provide the schedule which will supply exact gas costs values. We cache this code in the storage saving the schedule version.
- before running contract code we check if the cached code has the schedule version that is equal to the current saved schedule. If it is equal then run the code, if it isn't reinstrument with the current schedule.
- (refering to the code in question) when we update the schedule we want it to have stricly greater version than the current saved one: this guarantees that every instrumented contract code in cache cannot have the version equal to the current one. Thus, before executing a contract it should be reinstrument with new schedule.

If we allow to save a schedule with the version that equal to the current one, then we can end up in a situation when we execute cached code which was instrumented with gas costs that is not equal to the current gas costs in the current schedule.

Reply:
Sorry, I made a mistake. You are correct with your code :)

Reply:
@bkchr Phew! 😅Thanks for discussion anyway! I think I should have put something like this somewhere in the codebase in the first place! 

@gnunicorn merging your suggestion, thanks!

Reply:
Crammed this description to code_cache module!